File: Leetcode_11.py
#Time Complexdity: O(n)
#Space Complexity: O(1)

class Solution:
    def maxArea(self, height) -> int:
        def amountWater(i, j):
            return min(height[i], height[j]) * (j - i)

        maxWater = 0
        i, j = 0, len(height) - 1
        while i < j:
            if maxWater < amountWater(i, j):
                maxWater = amountWater(i, j)
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        return maxWater

File: LeetCode11.java
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach

class Solution {
    public int maxArea(int[] height) {
        int max=0;
        int n = height.length;
        int l=0, r =n -1;

        while(l<r){
            max= Math.max(max, Math.min(height[l], height[r])*(r-l));
            if(height[l]<height[r]) l++;
            else r--;
        }
       return max; 
    }
}